sdk/core/core-http/src/nodeFetchHttpClient.ts

```diff
@@ -88,8 +83,9 @@ export class NodeFetchHttpClient extends FetchHttpClient {
   }
 
   // eslint-disable-next-line @azure/azure-sdk/ts-apisurface-standardized-verbs
-  async fetch(input: CommonRequestInfo, init?: RequestInit): Promise<Response> {
-    return fetch(input, init);
+  async fetch(input: CommonRequestInfo, init?: CommonRequestInit): Promise<CommonResponse> {
+    const response = await node_fetch(input, init);
+    return (response as unknown) as CommonResponse;
```

Comment:
Minor but could this be reduced to just:
```suggestion
    return node_fetch(input, init) as unknown as Promise<CommonResponse>;
```